packages/ingest/util.js

```diff
@@ -1,6 +1,16 @@
 'use strict';
 
 const mime = require('mime-types');
+const KMS = require('@cumulus/aws-client/KMS');
+const { S3KeyPairProvider } = require('@cumulus/common/key-pair-provider');
+
+const decrypt = async (ciphertext) => {
+  try {
+    return await KMS.decryptBase64String(ciphertext);
```

Comment:
So we try base 64 decryption otherwise try to decrypt with the S3 keypair that was uploaded during deployment? Does the S3 keypair ever come into play for basic auth?

Reply:
S3 Keypair stuff is legacy support, shouldn't really come into play. The meat of this change is DRYing up this function between the HTTP/STP/FTP provider clients.